packages/web/src/App.tsx

```diff
@@ -0,0 +1,123 @@
+import { useQuery } from "@tanstack/react-query";
+import { useEffect, useState } from "react";
+import { ExportButton } from "./components/export-button";
+import { LogTable } from "./components/log-table";
+import { Pagination } from "./components/pagination";
+import { ServerFilter } from "./components/server-filter";
+import { SessionFilter } from "./components/session-filter";
+import type { LogEntry } from "./lib/api";
+import { api } from "./lib/api";
+import { useHandler } from "./lib/use-handler";
+import { getLogKey } from "./lib/utils";
+
+function App() {
+  const [serverName, setServerName] = useState<string | undefined>();
+  const [sessionId, setSessionId] = useState<string | undefined>();
+  const [oldestTimestamp, setOldestTimestamp] = useState<string | undefined>();
+  const [allLogs, setAllLogs] = useState<LogEntry[]>([]);
+  const [selectedIds, setSelectedIds] = useState<Set<string>>(new Set());
+
+  const { data, isLoading, error } = useQuery({
+    queryKey: ["logs", serverName, sessionId, oldestTimestamp],
+    queryFn: () =>
+      api.getLogs({
+        serverName,
+        sessionId,
+        limit: 100,
+        before: oldestTimestamp,
+        order: "desc",
+      }),
+  });
+
+  // Update accumulated logs when data changes
+  // biome-ignore lint/correctness/useExhaustiveDependencies: oldestTimestamp is read but intentionally excluded to avoid re-rendering on pagination
+  useEffect(() => {
```

Comment:
maybe `useInfiniteQuery` instead for this? https://tanstack.com/query/v5/docs/framework/react/guides/infinite-queries